packages/remix-server-runtime/server.ts

```diff
@@ -273,24 +273,21 @@ async function handleDocumentRequest({
   let routeModules = createEntryRouteModules(build.routes);
 
   let matchesToLoad = matches || [];
+
+  // get rid of the action, we don't want to call it's loader either
+  // because we'll be rendering the error/catch boundary, if you can get
+  // access to the loader data in the error/catch boundary then how the heck
+  // is it supposed to deal with thrown responses and/or errors in the loader?
   if (appState.catch) {
     matchesToLoad = getMatchesUpToDeepestBoundary(
-      // get rid of the action, we don't want to call it's loader either
-      // because we'll be rendering the catch boundary, if you can get access
-      // to the loader data in the catch boundary then how the heck is it
-      // supposed to deal with thrown responses?
-      matchesToLoad.slice(0, -1),
+      matchesToLoad,
       "CatchBoundary"
-    );
+    ).slice(0, -1);
```

Comment:
We don't want to `slice` _and then_ look for the boundary, otherwise we miss leaf boundaries.  Instead, find the boundary then `slice` so we don't run the loader for the boundary route.